river_torch/anomaly/scaler.py

```diff
@@ -0,0 +1,249 @@
+        Returns
+        -------
+        Anomaly score. A high score is indicative of an anomaly. A low score corresponds a
+        normal observation.
+        """
+        raw_score = self.anomaly_detector.score_one(*args)
+        min = self.min.update(raw_score).get()
+        max = self.max.update(raw_score).get()
+        score = (raw_score - min) / (max - min)
+
+        return score
+
+    def score_many(self, *args) -> float:
+        """Return scaled anomaly score based on raw score provided by the wrapped anomaly detector.
+
+        A high score is indicative of an anomaly. A low score corresponds to a normal observation.
+        Parameters
+        ----------
+        args
+            Depends on whether the underlying anomaly detector is supervised or not.
+        Returns
+        -------
+        Anomaly score. A high score is indicative of an anomaly. A low score corresponds a
+        normal observation.
+        """
+        raw_scores = self.anomaly_detector.score_many(*args)
+        for raw_score in raw_scores:
+            self.min.update(raw_score)
+            self.max.update(raw_score)
+
+        min = self.min.get()
+        max = self.max.get()
+        score = (raw_score - min) / (max - min)
+
+        return score
+
+
+class RollingStandardScaler(StandardScaler):
+    """Wrapper around an anomaly detector that standardizes the model's output using rolling mean and variance metrics.
+
+    Parameters
+    ----------
+    anomaly_detector
+    window_size
+    with_std : bool
+    """
+
+    def __init__(self, anomaly_detector, window_size=250, with_std=True):
+        super().__init__(anomaly_detector=anomaly_detector)
+        self.window_size = window_size
+        self.mean = RollingMean(window_size=window_size)
+        self.sq_mean = RollingMean(window_size=window_size) if with_std else None
+        self.with_std = with_std
+
+
+class AdaptiveStandardScaler(StandardScaler):
+    """Wrapper around an anomaly detector that standardizes the model's output using exponential running mean and variance metrics.
+
+    Parameters
+    ----------
+    anomaly_detector
+    alpha
+    with_std
+    """
+    def __init__(self, anomaly_detector, alpha=0.3, with_std=True):
+        super().__init__(anomaly_detector=anomaly_detector)
+        self.alpha = alpha
+        self.mean = EWMean(alpha=alpha)
+        self.sq_mean = EWMean(alpha=alpha) if with_std else None
+        self.with_std = with_std
+
+
+class RollingMinMaxScaler(MinMaxScaler):
+    """Wrapper around an anomaly detector that scales the model's output to $[0, 1]$ using rolling min and max metrics.
+
+    Parameters
+    ----------
+    anomaly_detector
+    window_size
+    """
+    def __init__(self, anomaly_detector, window_size=250):
+        super().__init__(anomaly_detector=anomaly_detector)
+        self.window_size = window_size
+        self.min = RollingMin(window_size=window_size)
+        self.max = RollingMax(window_size=window_size)
```

Comment:
Yes, that's a good point. One class for each scaling approach would be more concise. As for the API I think it would be nice to be able to use the scalers as a pipeline step, although I think that the pipeline implementation doesn't really allow that.

Reply:
We were talking about it today. Currently, the Autoencoded Anomaly Detector has the following structure:
![Picture 1](https://user-images.githubusercontent.com/42866082/177870404-79266baf-6da4-4728-8487-2a0982772f9d.png)
The Post Processor scales the scoring values and is followed by a Thresholder similar to the ones in river. 
Do you have an idea how one could implement that?

Reply:
I don't have a good answer yet. My initial plan was that every anomaly detector should be responsible of outputting a score which is in [0, 1]. Rescaling is ok, but it's a bit weird when you did it online.

Reply:
> @lucasczz We could add a Sigmoid post processing step to every AE?

That wouldn't be ideal, since most loss functions produce values > 0, which would mean that sigmoid would output values > 0.5.